### PR TITLE
Add documentation on tmt & regular expressions

### PR DESCRIPTION
**Pull request by happz**
Fixes https://github.com/teemtee/tmt/issues/2602.

The preview of tmt & regular expression would be at https://tmt--2646.org.readthedocs.build/en/2646/overview.html#regular-expressions

Pull Request Checklist

* [ ] write the documentation
* [ ] update the specification

### CODE REVIEW

**Comment by happz** [comment]
@psss @lukaszachy @thrix starting with `tmt * ls`, what other options would be using regular expressions? I don't mind adding pointers to regex info as we discover places where it's needed, especially while polishing plugin docstrings and options for the "render from docstrings" mini project, but if there are obvious options, maybe not based in plugins, that deserve a link, please, point them out.

File: docs/overview.rst [hunk 1]
```diff
@@ -591,6 +591,51 @@ example, an interactive mode would be enabled in this run::
...
+    * on the other hand, the "search" mode is more similar to
+      "contains" approach, and allows pattern to match anywhere in
+      the string. In this mode, ``foo.ar`` would match both
+      ``foobar`` and ``/foobar``.
+
+    In both modes, any characters may follow the matching pattern,
+    e.g. regular expression ``foo.bar`` is the same as
+    ``foo.bar.*``.
+
+    tmt sticks to the "match" mode by default. But, for historical
```
Comment (lukaszachy) [inline docs/overview.rst:629]: Isn't it the other way around? Which user facing regexp is used in 'match'?

Comment (happz) [reply docs/overview.rst:629]: That might have been my impression only, I admit. Or maybe a faithful wish. Let me switch it then.

Comment (happz) [reply docs/overview.rst]: Fixed in 64e76228

**Review by therazix — APPROVED** [review]

File: docs/overview.rst [hunk 2]
```diff
@@ -591,6 +591,51 @@ example, an interactive mode would be enabled in this run::
       $ TMT_PLUGIN_DISCOVER_FMF_VERBOSE=2 tmt run -a discover -h fmf ...
 
 
+.. _regular-expressions:
+
+Regular expressions
```
Comment (psss) [inline docs/overview.rst:596]:
```suggestion
Regular Expressions
```

File: docs/overview.rst [hunk 3]
```diff
@@ -591,6 +591,51 @@ example, an interactive mode would be enabled in this run::
...
+
+Python implementation
+    Since tmt is implemented in Python, Python's `re`__ package is
+    used for handling regular expressions. It comes with Python
+    standard library, and it is widely used and well documented.
+    For the deep dive into supported syntax, see
+    `Regular Expression Syntax`__ section, for a gentler
+    introduction there is a `Regular Expression`__ HOWTO.
+
+Regular expression must match at the beginning of the string
```
Comment (psss) [inline docs/overview.rst:612]: The heading should probably be updated as well to match the search default. What about this?

```suggestion
Search versus match
```

**Review by psss — CHANGES_REQUESTED** [review]
Just two minor things.

**Review by psss — APPROVED** [review]
Thanks for documenting this!

**Pull request merged**